### PR TITLE
#81 Add conformance rule reordering

### PR DESCRIPTION
**Pull request by GeorgiChochov**

### CODE REVIEW

**Comment by lokm01** [comment]
This doesn't seem to check the consistency of other rules. Is that coming in another PR?

**Comment by Zejnilovic** [comment]
> This doesn't seem to check the consistency of other rules. Is that coming in another PR?

There are no validations currently in general. When we get validations functions, they should be called before each. So maybe do validation in a separate PR instead of ad-hoc ifs?

**Comment by GeorgiChochov** [comment]
I'll stick it into this one.

**Comment by Zejnilovic** [comment]
When changing the order, the audit trail is really hard to read. Is this automatic or can we change it a bit?

image

**Comment by lokm01** [comment]
> When changing the order, the audit trail is really hard to read. Is this automatic or can we change it a bit?
> 
> image

It's automatic. There's an issue for making them more readable.

**Comment by GeorgiChochov** [comment]
I'll get back to adding validation to this once #461 is merged.

**Comment by Zejnilovic** [comment]
We have the same error message regardless of the direction of movement. Meaning, if I have dependency A -> B (B needs A to exist) and try moving A below B, it will tell me `Unable to move conformance rule as it depends on column "A"`, which would mean A is self-dependent.

Can we change the text for arrow down?

File: menas/ui/service/RuleService.js [hunk 1]
```diff
@@ -48,17 +48,21 @@ class RuleService {
...
       });
+
     const schemas = [schema];
     try {
       SchemaManager.updateTransitiveSchemas(schemas, newRules);
     } catch (e) {
-      return new InvalidResult(`Unable to move conformance rule as it depends on column "${e.fieldPath}"`)
+      const smallerIndex = Math.min(ruleIndex, otherRuleIndex);
+      const largerIndex = Math.max(ruleIndex, otherRuleIndex);
+      return new InvalidResult(`Unable to move conformance rule as it would break the dependency on column "${e.fieldPath}" between rule ${smallerIndex} and ${largerIndex}`)
```
Comment (Zejnilovic) [inline menas/ui/service/RuleService.js]: index + 1 for the message and rules headings.

Comment (GeorgiChochov) [reply menas/ui/service/RuleService.js]: Fair point, thanks for keeping my laziness in check.

**Pull request merged**